src/breaking-change.ts

```diff
@@ -162,13 +165,183 @@ async function runOad(oldSpec: string, newSpec: string) {
   return JSON.parse(result);
 }
 
-//main function
-export async function runScript() {
-  console.log(`ENV: ${JSON.stringify(process.env)}`);
-  // Used to enable running script outside TravisCI for debugging
-  const isRunningInTravisCI = process.env.TRAVIS === "true";
+type SwaggerVersionType = "preview" | "stable";
+type RuntimeError = { error: Error; old: string; new: string };
+type SwaggerMetaData = {
+  version: string;
+  versionType: SwaggerVersionType;
+  fileName: string;
+  folder: string;
+};
 
-  /**
+function appendException(errors: RuntimeError[]) {
+  const errorResult: format.MessageLine = errors.map((it) => ({
+    type: "Raw",
+    level: "Error",
+    message: "Runtime Exception",
+    time: new Date(),
+    extra: {
+      new: it.new,
+      old: it.old,
+      details: utils.cutoffMsg(it.error.stack) || "",
+    },
+  }));
+
+  fs.appendFileSync("pipe.log", JSON.stringify(errorResult) + "\n");
+  console.log(`oad error log: ${JSON.stringify(errorResult)}`);
+}
+export class SwaggerVersionManager {
+  getRPFolder(swaggerFile: string) {
+    // Needs to consider other cases
+    const segments = swaggerFile.split(/\\|\//)
+    if (segments && segments.length > 3) {
+      return segments.slice(0, -3).join("/");
+    }
+    return undefined
+  }
+
+  getAllSwaggers(folder: string) {
+    const pathPattern = path.join(folder, "**/*.json");
+    return glob.sync(pathPattern, {
+      ignore: [
+        "**/examples/**/*.json",
+        "**/quickstart-templates/*.json",
+        "**/schema/*.json",
+      ],
+    });
+  }
+
+  getVersionMapping(swaggerFile: string) {
+    const swaggerMetaData: SwaggerMetaData[] = [];
+    const folder = this.getRPFolder(swaggerFile);
+    if (!folder) {
+      return swaggerMetaData
+    }
+    const allSwaggers = this.getAllSwaggers(folder);
+    for (const swagger of allSwaggers) {
+      const version = getVersionFromInputFile(swagger);
+      const fileName = path.basename(swagger);
+      // Needs to consider other cases
+      const versionType = swagger.includes("preview") ? "preview" : "stable";
```

Comment:
Should apiVersion be examined with 'privatepreview'

Reply:
fixed